### ISSUE TITLE
eventColor - access line info

### ISSUE THREAD

**Issue by chrishopkins** [marmelab/EventDrops#49]
Is there a way of accessing the event line name when inspecting a datum within eventColor?

I am plotting different types of dates within a row and I would like to colour each datum based upon what type of date they are.

I have hacked around this by cycling through the data in all event lines for a given date type (within .eventColor) and returning the specific colour for that date type if the datum value matches. This breaks if the same date value exists elsewhere on another line, as the same colour will get assigned to both irrespective of data type.

An improvement would be to look when index becomes zero, as this will signify that the datum is on the next line to the previous and from this I can work out what line it belonged to. 

I was wondering however if there is a nicer way of doing this, if there is a way of getting the line name for a datum within eventColor?

Or to be able to pass in multiple lists to the createEvent function and colour a datum based upon which list it came from?

**Comment by jpetitcolas** [comment]
Sorry for the delay. 

If I understand you correctly, you want to be able to access some extra information from the passed data in every configuration functions? Would something like:

``` js
var data = [
    { name: 'Some data', data: [
        { date: '2014-05-02', foo: 'bar', extra: 'Lorem Ipsum' },
        // ...
    ] }
];

var chart = d3.chart.eventDrops()
    .eventColor(function (d) {
        return d.foo === 'bar' ? 'red' : 'green';
    })
    .eventClick(function (d) {
        console.log(d.extra);
    });
```
Fulfill your need?

**Comment by GeorgeL9** [comment]
@jpetitcolas I am curious about the eventClick event. I am having difficulties with it firing from your example:
```

var eventDropsChart = d3.chart.eventDrops().start(new Date('Jan 1, 2015'))
    .eventClick(function (d) {
        console.log(d);
    });
```

**Comment by jpetitcolas** [comment]
Just to make things clear: the snippet above is not currently implemented, but is an idea for a future implementation. Currently, `d` should only be the drop date. 

We fixed recently event handlers. I've just rebuilt library. Can you retry with last master version?

**Comment by GeorgeL9** [comment]
+1, eventClick and eventHover is now working!

I am building a tooltip so I can inspect events, is this feature in the official roadmap?

**Comment by jpetitcolas** [comment]
Yes, I think we are going to embed it in a 0.2.1 or 0.3 version (depending if there are some backward compatibility breaks). For now, we are focusing on delivering 0.2.0. :)

**Comment by GeorgeL9** [comment]
Sorry if this is the wrong place to ask, but is it possible to get the index and category on the eventHover event?

> var data = [
>         {
>             name: "Stars",
>             dates: [new Date('2014/09/15 13:24:54'), new Date('2014/09/15 13:25:03')],
>             repos: ['test/test','debug/debug']
>         }, {
>             name: "Pushs",
>            dates: [new Date('2014/08/15 11:24:54'), new Date('2014/09/15 13:25:03')],
>             repos: ['test/test','debug/debug']
>         }];

Ideally even, if I could just get the repo name.

**Comment by jpetitcolas** [comment]
Can you just open a new issue to not pollute the original one? :)

**Comment by anicke** [comment]
Pull request #76 should enable this kind of input data and functionality.

```
var data = [
    { name: 'Some data', data: [
        { date: '2014-05-02', foo: 'bar', extra: 'Lorem Ipsum' },
        // ...
    ] }
];

var chart = d3.chart.eventDrops()
    .dateAttribute("date")
    .eventColor(function (d) {
        return d.foo === 'bar' ? 'red' : 'green';
    })
    .eventClick(function (d) {
        console.log(d.extra);
    });

```

**Comment by ChaitanyaSharma** [comment]
I merged this request into my master repository since I want to get additional data. However, my the data does not render using the example you mentioned. Could you help me figure out what I could be doing incorrect here?

demo.js file attached.
[demo.js.zip](https://github.com/marmelab/EventDrops/files/248054/demo.js.zip)

**Comment by oddieg** [comment]
Is there an rough ETA on a release for the above fix?

**Comment by jpetitcolas** [comment]
@oddieg: We'll release a `0.3` version once current four PR (up to #80) are merged.

**Comment by jpetitcolas** [comment]
Should be fixed by #76.
